randovania/game_connection/executor/am2r_executor.py

```diff
@@ -5,23 +5,20 @@
 import logging
 import struct
 import typing
-from asyncio import StreamReader, StreamWriter
 from enum import IntEnum
 
 from PySide6.QtCore import QObject, Signal
 
+from randovania.game_connection.executor.common_socket_holder import CommonSocketHolder
+
 
 class AM2RConnectionException(Exception):
     pass
 
 
-@dataclasses.dataclass()
-# TODO: AM2R, Dread and future MSR have here very similar attribute. Refactor this to a common SocketHolder
-class AM2RSocketHolder:
-    reader: StreamReader
-    writer: StreamWriter
+@dataclasses.dataclass(frozen=True)
+class AM2RSocketHolder(CommonSocketHolder):
     api_version: int
-    buffer_size: int
```

Comment:
wrong line lol

Reply:
quite. removing the buffer size was intentional, am2r's executor does not use it whatsoever. Twas the classic copy paste from dread.